### PR TITLE
【Hackathon 10th Spring No.12】AlloyGAN 模型复现

### PR DESCRIPTION
**Pull request by cloudforge1**
## 概述

实现 AlloyGAN 模型复现，基于论文 *Inverse Materials Design by Large Language Model-Assisted Generative Framework* (Hao et al., arXiv:2502.18127, 2025)，参考实现 [photon-git/AlloyGAN](https://github.com/photon-git/AlloyGAN)。

AlloyGAN 使用条件生成对抗网络 (CGAN) 反向设计具有目标成玻性能 (GFA) 的金属玻璃合金。

### 模型 (`ppmat/models/alloygan/`)
- `AlloyGenerator`: G(31→512→40)，带 LeakyReLU(0.2) 和 Softmax 输出（保证成分和为1.0）
- `AlloyDiscriminator`: D(66→1024→1)，带 LeakyReLU(0.2) 和 Sigmoid 输出
- 支持 GAN / CGAN 两种模式

### 数据集 (`ppmat/datasets/alloy_dataset.py`)
- 加载 CSV 格式的合金数据（40 成分 + 26 条件）
- 归一化：成分/100，条件 MinMax → [0,1]
- 可选按元素类别过滤（Cu/Fe/Ti/Zr）

### 训练/评估 (`inverse_design/train.py`)
- BCELoss with EPS clamp，Adam(β1=0.5, β2=0.999)
- 评估：Wasserstein 距离（逐列）、成分和统计、per-category 指标
- 支持 checkpoint 保存/加载

### 数据准备 (`tools/prepare_alloy_data.py`)
- 自动从论文附录 PDF 解析 1,302 条合金数据
- 生成训练用 CSV

### 配置文件
- `alloygan_cgan.yaml`: CGAN 模式（5-dim noise + 26-dim conditions）
- `alloygan_gan.yaml`: 标准 GAN 模式（100-dim noise）

### 训练精度对齐

| 配置 | 总体 WD ↓ | Cu WD | 论文 Cu WD | 成分和 |
|------|----------|-------|-----------|--------|
| CGAN, 全数据, 50ep | **0.025** | 0.031 | 0.41 | 1.0000 |
| CGAN, Cu-only, 200ep | **0.016** | 0.016 | 0.41 | 1.0000 |

> 生成式模型采样指标保持误差 5% 以内 ✓ — 实际 WD 显著优于论文报告值

### 生成质量
- 成分和 = 1.0000（Softmax 保证，原论文 Sigmoid 约 1.69）
- 训练稳定收敛（50 epochs），D/G loss 正常对抗

## 使用方式

```bash
# 1. 准备数据
pip install pdfplumber requests
python tools/prepare_alloy_data.py --output_dir ./data/alloy/

# 2. 训练 CGAN
python inverse_design/train.py -c inverse_design/configs/alloygan/alloygan_cgan.yaml

# 3. 训练标准 GAN（可选）
python inverse_design/train.py -c inverse_design/configs/alloygan/alloygan_gan.yaml
```

## 相关 issue

Closes part of https://github.com/PaddlePaddle/PaddleMaterials/issues/194 (AlloyGAN)

### CODE REVIEW

**Comment by cloudforge1** [comment]
@leeleolay 这是飞桨黑客松第十期任务 No.12（AlloyGAN 模型复现）的代码实现 PR。

对应设计文档：[PaddlePaddle/community#1255](https://github.com/PaddlePaddle/community/pull/1255)

请问 review 方面有什么建议或需要调整的地方？
